src/Http/Http.Results/src/Results.cs

```diff
@@ -667,15 +697,24 @@ public static IResult Problem(ProblemDetails problemDetails)
 
         problemDetails.Title = title ?? problemDetails.Title;
 
+        CopyExtensions(extensions, problemDetails);
+
+        return TypedResults.Problem(problemDetails);
+    }
+
+    private static void CopyExtensions(IDictionary<string, object?>? extensions, HttpValidationProblemDetails problemDetails)
+    {
         if (extensions is not null)
         {
             foreach (var extension in extensions)
             {
+#pragma warning disable IL2026 // Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code
```

Comment:
I'm not sure why this is valid. Why can we disable these warnings?

In dotnet/runtime we had the suppressions in a .xml file until we addressed them. That way we could view all the unsolved suppressions in a single place, and didn't need to edit the code until it was addressed.

Reply:
Looking at it more - don't we need to warn the caller that the objects in `extensions` aren't guaranteed to serialize correctly?

Reply:
The PR description talks about it. These warnings are being removed with https://github.com/dotnet/aspnetcore/pull/45886. The suppression is temporary.